Prolangle/Extensions/LanguageEnumerablePickExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using Prolangle.Languages.Framework;
+using Prolangle.Services;
+using Weighted_Randomizer;
+
+namespace Prolangle.Extensions;
+
+public static class LanguageEnumerablePickExtensions
+{
+	public static ILanguage PickWeightedRandom(this IEnumerable<ILanguage> languages,
+		GameSeeder seeder)
+	{
+		var randomizer = new StaticWeightedRandomizer<ILanguage>(seeder.Seeder());
+
+		foreach (ILanguage language in languages)
+		{
+			// we weigh a language's TIOBE rating against frequency in
+			// which the language should appear
+			var weight = language.TiobeRating is null or 0 ? 100 : 100 / language.TiobeRating;
```

Comment:
Maybe scale so that `languages.Min(l => l.TiobeRating)` equals `99` (i.e. `1` less than the unknown/unranked languages)?

Reply:
I've done both variants in [6c6c466](https://github.com/ricardoboss/Prolangle/pull/123/commits/6c6c466169baf5a6dd2d272f628324703ab72dd5) and [2732af4](https://github.com/ricardoboss/Prolangle/pull/123/commits/2732af4d82bbdff42083b336b681c814b8886dff).